src/Bicep.Core/Semantics/ParamsSemanticModel.cs

```diff
@@ -15,26 +19,74 @@ public class ParamsSemanticModel
         public Compilation? BicepCompilation { get; }
         public ParamsTypeManager ParamsTypeManager { get; }
         public ParamsSymbolContext ParamsSymbolContext { get; }
+        public List<IDiagnostic> allDiagnostics { get; } = new();
+        public IFileResolver fileResolver { get; }
         
-        public ParamsSemanticModel(BicepParamFile bicepParamFile, Compilation? bicepCompilation = null)
+        public ParamsSemanticModel(BicepParamFile bicepParamFile, IFileResolver fileResolver, Func<Uri, Compilation>? getCompilation = null)
         {
+            this.fileResolver = fileResolver;
+
+            //parse using statement and link bicep template 
+            this.BicepParamFile = bicepParamFile;
+
+            Uri? bicepFileUri = TryGetBicepFileUri();
+            if(bicepFileUri is {} && getCompilation is {})
+            {   
+                this.BicepCompilation = getCompilation(bicepFileUri);
+            }
+
+            //binder logic
             var paramsSymbolContext = new ParamsSymbolContext(this);
             ParamsSymbolContext = paramsSymbolContext;
             this.BicepParamFile = bicepParamFile;
-            this.BicepCompilation = bicepCompilation;
             this.ParamBinder = new(bicepParamFile, paramsSymbolContext); 
             this.ParamsTypeManager = new(this, ParamBinder);
             // name binding is done
             // allow type queries now
             paramsSymbolContext.Unlock();
         }
 
-        public IEnumerable<IDiagnostic> GetDiagnostics()
-            => BicepParamFile.ProgramSyntax.GetParseDiagnostics();
+        public IEnumerable<IDiagnostic> GetAllDiagnostics()
+            => this.allDiagnostics.Concat(BicepParamFile.ProgramSyntax.GetParseDiagnostics());
         
         /// <summary>
         /// Gets the file that was compiled.
         /// </summary>
         public ParamFileSymbol Root => this.ParamBinder.ParamFileSymbol;
+
+        private Uri? TryGetBicepFileUri()
+        {
+            //TODO: consolidate this into one method
+            var usingDeclarations = BicepParamFile.ProgramSyntax.Children.OfType<UsingDeclarationSyntax>();
+
+            if(usingDeclarations.Count() == 0)
+            {
+                this.allDiagnostics.Add(new DiagnosticBuilder.DiagnosticBuilderInternal(new TextSpan(0, 0)).UsingDeclarationNotSpecified());
+                return null;
+            }
+            
+            if(usingDeclarations.Count() > 1)
+            {
+                this.allDiagnostics.Add(new DiagnosticBuilder.DiagnosticBuilderInternal(new TextSpan(0, 0)).MoreThenOneUsingDeclarationSpecified());
+                return null;
+            }
+
+            if(!PathHelper.TryGetUsingPath(usingDeclarations.FirstOrDefault()!, out var bicepFilePath, out var failureBuilder))
```

Comment:
As discussed, please avoid using `!` to convert a nullable value to a non-null value in product code (test code is fine!). The reason for this is that it's easy to miss in refactors, and can result in bugs slipping in.

Here I think the best way to avoid it would be to change the check for 0 `using` declarations to:
```csharp
if (usingDeclarations.FirstOrDefault() is not {} usingDeclaration)
{
   ...
}
```

You'll then be able to replace `usingDeclarations.FirstOrDefault()!` after this check with just `usingDeclaration`